### DISCUSSION TITLE
Initializing CefSharp inside of a third-party application causes a crash

### DISCUSSION THREAD

**Discussion by dave-lewis73** [cefsharp/CefSharp#4480] [General]
### Is there an existing issue for this?

- [X] I have searched both open and closed issues

### CefSharp Version

112.2.270

### Operating System

Windows 10

### Architecture

x86

### .Net Version

.NET 4.5

### Implementation

WinForms

### Reproduction Steps

I have been developing an extension to ArcMap (a third party mapping application) and have been using CefSharp as my web browser within the extension, since I can't get the out-of-the-box WebBrowser in Visual Studio 2015 to work consistently.  I have four web browser controls that I have on a WinForm that each have a link to a different web map (URL).  Everything works fine on my development machine, but the parent application (ArcMap) is crashing on each machine that I have deployed the solution to.  It appears to happen immediately when I start up the extension, which is where CefSharp is initialized.  I am calling Cef.InitializeAsync(new CefSettings()).  As an fyi, the Microsoft Visual C++ 2015-2019 Redistributable has been installed on the machines.

### Expected behavior

I would expect that CefSharp would initialize properly and that I would see four web maps (one in each of my CefSharp browsers.

### Actual behavior

The parent application (ArcMap) crashes on deployed machines when attempting to initialize CefSharp with no displayed error message.

### Regression?

I have not tested in a previous version of CefSharp.

### Known Workarounds

None.

### Does this problem also occur in the CEF Sample Application

Not Tested

### Other information

None.

**Comment by aimcosoftware** [comment]
Check Windows logs for any errors, but you've almost certainly not deployed something.

**Comment by dave-lewis73** [comment]
The Windows logs do not indicate any issues.  As an fyi, I have downloaded the test app and executed it on one of the deployment machines with no problem.  That said,  I compared the files that are bundled with the test app and those that I am deploying and nothing is missing.  I am deploying all of the files that are bundled with the test app.

**Reply by aimcosoftware** [reply]
> I have downloaded the test app and executed it on one of the deployment machines with no problem

Surely that says that it's not CEFSharp/CEF at fault?

**Comment by amaitland** [comment]
Converted to discussion as this seems more like a support request than an actual bug that needs fixing.

> Operating System
Windows 10

Are all your target machines also using `Windows 10`?

> .NET 4.5 

`CefSharp` hasn't supported `.Net 4.5` for quite a few years. Are you using `.Net 4.5.2`?

> Everything works fine on my development machine, but the parent application (ArcMap) is crashing on each machine that I have deployed the solution to. It appears to happen immediately when I start up the extension, which is where CefSharp is initialized.

If it works within `ArcMap` on your dev machine then it's very unlikely what you are seeing is a bug in `CefSharp`.

There are a few important things to remember though:
- Only a single `CEF` version can be loaded per process. If another `extension` is already using `CefSharp` then you'd need to match their version or host the browser in a different process.
- `CefSharp` is written in `VC++` which only supports running on the `Default AppDomain`. Some plugin systems like to load dlls in different `AppDomains` which will result in an exception.

> the Microsoft Visual C++ 2015-2019 Redistributable has been installed on the machines.

Have you installed the correct architecture? You'd need to make sure the `x86` version was installed. 

> The Windows logs do not indicate any issues

Check the `CEF` log file also. https://github.com/cefsharp/CefSharp/wiki/Trouble-Shooting#log-file
Make sure to set `LogFile` to somewhere writable.

Does `ArcMap` have some sort of log file? Without an error message/exception it's pretty much impossible to know what's going on.
